### ISSUE TITLE
Can not edit or create new records in rails admin

### ISSUE THREAD

**Issue by mihaj** [sferik/rails_admin#2830]
I have a problem with crating and updating records with rails_admin. For update I get the 404 error, where POST actions is not found:
`{"status":404,"error":"Not Found","exception":"#\u003cActionController::RoutingError: No route matches [POST] \"/admin/api_key/1/edit\"....`

But if i look in the routes file, i see it's registered at PUT method.

For creating new record i get:
`{"status":422,"error":"Unprocessable Entity","exception":"#\u003cActionController::InvalidAuthenticityToken: ActionController::InvalidAuthenticityToken\u003e","traces":{"Application Trace":[],"Framewor....`

my config file:
`RailsAdmin.config do |config|
  config.actions do
    dashboard                     # mandatory
    index                         # mandatory
    new
    export
    bulk_delete
    show
    edit
    delete
    show_in_app
  end
end`

my route setting:
`Rails.application.routes.draw do
  mount RailsAdmin::Engine => '/admin', as: 'rails_admin'`

**Comment by zekedran** [comment]
I'm getting the same too, and I think this happened after I upgraded my gems

**Comment by zekedran** [comment]
Yep, the problem is on `1.1.1`; I reverted back to `1.1.0` and no errors encountered :)
But there's a need to find why its happening in the new version!

**Comment by mshibuya** [comment]
Not reproducible here, can you make the problem happen by using brand-new Rails app?

**Comment by ss44** [comment]
Just installed a fresh rails instance, and was able to replicate same issue. Rolling back to 1.1.0 resolved it for me also.

Using papertrail and devise if that helps any?

**Comment by amicming** [comment]
I just did fresh rails instance with papertrail and devise and I can create new record. I'm not able to reproducible.

**Comment by siklodi-mariusz** [comment]
I am having the same issue with `1.1.1` and going back to `1.1.0` solves the problem but as stated [here](https://github.com/sferik/rails_admin#action-required-security-issue), it's recommended to upgrade to `1.1.1` because `1.1.0` is known to have CSRF vulnerability.

**Comment by juanibiapina** [comment]
I can also reproduce it with a fresh Rails app using --api.

**Comment by tinbka** [comment]
Also reported here: https://github.com/rails/rails/issues/21948#issuecomment-307165595
I doubt it's Rails core issue, though, as it would have been fixed in newer versions otherwise.

**Comment by juanibiapina** [comment]
The rails core issue seems to happens when you close the page and open again, so it might be unrelated.

**Comment by tinbka** [comment]
I've considered it, of course, as well as the secondary calls (e.g. pjax) that may be recreating the token, and two users accessing the same site with the same admin_user.id.  
I'll dig it deeper when I have time.

**Comment by tinbka** [comment]
Fixed this by one or both of:  
- upgrading to Rails 5.1  
- implementing nginx real_ip module so that `request.ip` is always equal to `request.remote_ip`

**Comment by juanibiapina** [comment]
I have rails 5.1.1 and no nginx.

**Comment by terry90** [comment]
@lesliev Dude, you totally are a day saver.
This solution works for both issues as advertised ! Thanks a lot.

**Comment by mshibuya** [comment]
Fixed by #3006.

**Comment by davideluque** [comment]
I had this issue and adding `config.middleware.use ::Rack::MethodOverride` to the application.rb did not work for me. 

This issue was present in my recently deployed production server (nginx+puma+rails), SSL enabled. 

I fixed this by adding `proxy_set_header X-Forwarded-Proto https;` to the Nginx site configuration, like this:

```
location @puma {
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header Host $http_host;
  proxy_set_header X-Forwarded-Proto https;
  proxy_redirect off;

  proxy_pass http://puma;
}
```
This also solved getting "connection not secure" in Chrome even though my certificate was valid. It's not a rails_admin but it was showing there.

**My settings**
Rails API only.
Rails 5.2.3
rails_admin 1.42
